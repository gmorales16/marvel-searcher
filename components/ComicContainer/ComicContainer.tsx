import {
  Container,
  Image,
  InfoContainer,
  Title,
  GeneralData,
  Description,
} from "./styledComicContainer";

export interface IComicContainer {
  image: string;
  title: string;
  published: string;
  writer: any;
  penciler: string;
  artist: string;
  description: string;
}

export default function ComicContainer({
  image,
  title,
  published,
  writer,
  penciler,
  artist,
  description,
}: IComicContainer) {
  return (
    <Container>
      <Image src={image} alt="Comic Image"></Image>
      <InfoContainer>
        <Title data-testid="comic-title">{title}</Title>
        <GeneralData>
          Published: {published} <br />
          Writer: {writer} <br />
          Penciler: {penciler} <br />
          Cover Artist: {artist} <br />
        </GeneralData>
        <Description data-testid="comic-description">{description}</Description>
      </InfoContainer>
    </Container>
  );
}
