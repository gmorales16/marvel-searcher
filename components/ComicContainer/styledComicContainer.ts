import styled from "styled-components";

const Container = styled.div`
  display: flex;
  justify-content: space-evenly;
  column-count: 2;
  margin: 5rem;

  @media (max-width: 768px) {
    flex-direction: column;
  }
`;

const Image = styled.img`
  width: 25%;
  @media (max-width: 768px) {
    width: 100%;
  }
`;

const InfoContainer = styled.div`
  color: #3e3e3e;
  letter-spacing: 0px;
  @media (max-width: 768px) {
    width: 80%;
  }
`;

const Title = styled.h1`
  font-size: 28px;
  font-weight: bold;
`;

const GeneralData = styled.h3`
  margin-top: 5rem;
  white-space: pre-wrap;
  font-size: 21px;
  @media (max-width: 768px) {
    margin-top: 2rem;
  }
`;

const Description = styled.p`
  margin-top: 5rem;
  width: 437px;
  font-size: 18px;
  @media (max-width: 768px) {
    width: 100%;
    margin-top: 2rem;
  }
`;

export { Container, Image, InfoContainer, Title, GeneralData, Description };
