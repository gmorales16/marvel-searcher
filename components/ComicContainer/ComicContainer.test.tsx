import "jsdom-global/register";
import "@testing-library/jest-dom/extend-expect";
import React from "react";
import { render, screen } from "@testing-library/react";
import ComicContainer, { IComicContainer } from "./ComicContainer";

describe("<ComicContainer/>", () => {
  test("renders ComicContainer Component", () => {
    const mockData: IComicContainer = {
      image: "Hi im Iron Man Link Image",
      title: "Iron Man",
      published: "16 / 12 / 2000",
      writer: "Carl Johnson",
      penciler: "Rohan Kishibe",
      artist: "Lionel Messi",
      description:
        "He launches missiles, flies and his suit is made of iron, that's why he is called Iron Man. Oh yes, he also shoots lightning from his chest",
    };
    render(<ComicContainer {...mockData} />);

    const imageElement = screen.getByAltText("Comic Image");
    const titleElement = screen.getByTestId("comic-title");
    const publishedElement = screen.getByText(/Published:/i);
    const writerElement = screen.getByText(/Writer:/i);
    const pencilerElement = screen.getByText(/Penciler:/i);
    const artistElement = screen.getByText(/Cover Artist:/i);
    const descriptionElement = screen.getByTestId("comic-description");

    expect(imageElement).toBeInTheDocument();
    expect(titleElement).toBeInTheDocument();
    expect(publishedElement).toBeInTheDocument();
    expect(writerElement).toBeInTheDocument();
    expect(pencilerElement).toBeInTheDocument();
    expect(artistElement).toBeInTheDocument();
    expect(descriptionElement).toBeInTheDocument();

    // Take a snapshot of the SearchBar component
    const { asFragment } = render(<ComicContainer {...mockData} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
