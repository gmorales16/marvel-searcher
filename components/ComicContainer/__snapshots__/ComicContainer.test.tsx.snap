// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ComicContainer/> renders ComicContainer Component 1`] = `
<DocumentFragment>
  <div
    class="ComicContainer__Container-sc-16s3lk1-0 kBKVGw"
  >
    <img
      alt="Comic Image"
      class="ComicContainer__Image-sc-16s3lk1-1 iGLqoP"
      src="Hi im Iron Man Link Image"
    />
    <div
      class="ComicContainer__InfoContainer-sc-16s3lk1-2 bqgqRw"
    >
      <h1
        class="ComicContainer__Title-sc-16s3lk1-3 kAzrHJ"
        data-testid="comic-title"
      >
        Iron Man
      </h1>
      <h3
        class="ComicContainer__GeneralData-sc-16s3lk1-4 WRtzq"
      >
        Published: 16 / 12 / 2000 
        <br />
        Writer: Carl Johnson 
        <br />
        Penciler: Rohan Kishibe 
        <br />
        Cover Artist: Lionel Messi 
        <br />
      </h3>
      <p
        class="ComicContainer__Description-sc-16s3lk1-5 ffpJvU"
        data-testid="comic-description"
      >
        He launches missiles, flies and his suit is made of iron, that's why he is called Iron Man. Oh yes, he also shoots lightning from his chest
      </p>
    </div>
  </div>
</DocumentFragment>
`;
